README.md

```diff
@@ -31,4 +31,73 @@ Po zamknięciu zapisów moderatorzy roku przypisują studentów do grup uwzględ
 - Może przydzielać role
 - Może dodawać, usuwać i edytować konta studentów
 
+
+# Typy encji
+
+## Student
+
+Reprezentuje aktora Student.
+- Imię
+- Nazwisko
+- E-mail
+- Numer indeksu
+
+## Moderator
+
+Podklasa typu encji _Student_
+
+Reprezentuje aktora Moderator.
+
+## Administrator
+
+Podklasa typu encji _Moderator_
+
+Reprezentuje aktora Administrator.
+
+## Rocznik
+
+Reprezentuje grupę użytkowników na tym samym roczniku i kierunku.
+
+- Rok rozpoczęcia nauki
+- Nazwa kierunku
+
+## Zapis
+
+Reprezentuje zapis, w ramach którego studenci będą przypisywani do grup.
+
+- Nazwa zapisu
+- Stan zapisu (otwarty, oczekiwanie na wyniki itp.)
+- Minimalna ilość punktów na przedmiot
+- Maksymalna ilość punktów na przedmiot
+- Maksymalna ilość punktów na grupę
+
+## Przedmiot
+
+Reprezentuje przedmiot, z którego organizowane są zajęcia.
+
+- Nazwa przedmiotu
+- Rodzaj zajęć (wykład, laboratoria itp.)
+- Sposób prowadzenia zajęć (zdalne, stacjonarne itp.)
+
+## Grupa
+
+Reprezentuje grupę zajęciową.
+
+- Prowadzący
+
+## Termin
+
+Reprezentuje blok zajęciowy odbywający się w danym terminie.
+
+- Dzień wykonywania
+- Godzina rozpoczęcia
+- Godzina zakończenia
+- Ograniczenia czasowe (tydzień A/B itp.)
```

Comment:
"Dzień wykonywania" zamieniłbym na "Dzień odbywania się zajęć".

Reply:
Przyjmujemy konwencję godziny startu i zakończenia czy godziny startu i czas trwania? Nie jestem pewny co się lepiej sprawdza, trzeba by było sprawdzić czy pg ma jakiś typ do timediffów.

Reply:
Ograniczenia czasowe też mi się wydają niedomodelowane.
Względnie możemy zamienić to pole na "Typ tygodnia" albo coś podobnego, ograniczając się tylko do A/B.

Możemy również pominąć tą złożoność typu tygodni w całości.

Reply:
To pozostawia problem typu 1 połowa semestru / 2 połowa semestru...
Wypadałoby coś tam jednak mieć, na pewno tydzień A/B, chociaż nie jestem pewien jak to zamodelować.
Czy dobrym pomysłem byłoby dodanie nowej encji, w której przechowywany byłby opis słowny co takie ograniczenie oznacza, np. "Tydzień A" lub "1 połowa semestru"? Nie do końca mi się to rozwiązanie podoba.

Co do timediffów, to jest w pg datediff, ale wygląda na to że trochę trudny w użyciu. Jest natomiast typ interval, którym łatwo można zamodelować czas trwania używając jedynie operatora "+", więc jest to lepsze rozwiązanie.

Reply:
Muszę się nad tym zastanowić, ale wprowadzenie pojęcia "1 połowa semestru" wiąże się z:
1. wprowadzeniem datowania planu
2. "tagowania" poszczególnych terminów, co byłoby rozróżnieniem wyłącznie graficznym

Myślę, że możemy na razie tą część pominąć i skupić się na reszcie.

Reply:
Zgoda.

Reply:
Jak na razie w takim razie pomijamy to problem.